build-kmod-zfs.sh

```diff
@@ -0,0 +1,62 @@
+#!/usr/bin/bash
+
+set -oeux pipefail
+
+
+ARCH="$(rpm -E '%_arch')"
+KERNEL="$(rpm -q kernel --queryformat '%{VERSION}-%{RELEASE}.%{ARCH}')"
+RELEASE="$(rpm -E '%fedora')"
+
+# allow pinning to a specific release series (eg, 2.0.x or 2.1.x)
+ZFS_MINOR_VERSION="${ZFS_MINOR_VERSION:-}"
+
+cd /tmp
+
+# Use cURL to fetch the given URL, saving the response to `data.json`
+curl "https://api.github.com/repos/openzfs/zfs/releases" -o data.json
+ZFS_VERSION=$(jq -r --arg ZMV "zfs-${ZFS_MINOR_VERSION}" '[ .[] | select(.prerelease==false and .draft==false) | select(.tag_name | startswith($ZMV))][0].tag_name' data.json|cut -f2- -d-)
+echo "ZFS_VERSION==$ZFS_VERSION"
+
+
+### zfs specific build deps
+rpm-ostree install libtirpc-devel libblkid-devel libuuid-devel libudev-devel openssl-devel libaio-devel libattr-devel elfutils-libelf-devel python3-devel libffi-devel libcurl-devel ncompress
+
+
+### BUILD zfs
+echo "getting zfs-${ZFS_VERSION}.tar.gz"
+curl -L -O https://github.com/openzfs/zfs/releases/download/zfs-${ZFS_VERSION}/zfs-${ZFS_VERSION}.tar.gz
+# no-same-owner/no-same-permissions required for F40 based images building on podman 3.4.4 (ubuntu 22.04)
+tar -z -x --no-same-owner --no-same-permissions -f zfs-${ZFS_VERSION}.tar.gz
+
+# patch the zfs-kmod.spec.in file for older zfs versions
+ZFS_MAJ=$(echo $ZFS_VERSION | cut -f1 -d.)
+ZFS_MIN=$(echo $ZFS_VERSION | cut -f2 -d.)
+ZFS_PATCH=$(echo $ZFS_VERSION | cut -f3 -d.)
+DO_PATCH=1
+if [ "${ZFS_MAJ}" -lt "2" ]; then
+    DO_PATCH=0
+elif [ "${ZFS_MAJ}" -eq "2" ] && [ "${ZFS_MIN}" -lt "2" ]; then
+    DO_PATCH=0
+elif [ "${ZFS_MAJ}" -eq "2" ] && [ "${ZFS_MIN}" -eq "2" ] && [ "${ZFS_PATCH}" -lt "3" ]; then
+    DO_PATCH=0
+fi
+if [ "${DO_PATCH}" -eq "0" ]; then
+    patch  -b -uN -i zfs-kmod-spec-in.patch zfs-${ZFS_VERSION}/rpm/generic/zfs-kmod.spec.in
+fi
```

Comment:
Same as the earlier comment, we don't need this whole patching section because the ZFS versions old enough to need the patch won't even compile on 6.8 or 6.9

Reply:
This is still in the Ucore one. As a heads up